src/ballet/wsample/fd_wsample.h

```diff
@@ -52,81 +52,119 @@ void *            fd_wsample_delete   ( void * shmem  );
 
    The hint can also specify whether sampling will be done with deleting
    or without deleting:
-   DELETE: Sampling will be done without replacement, i.e. mostly with
-       fd_wsample_sample_and_delete and/or
-       fd_wsample_sample_and_delete_many.
-   NODELETE: Sampling will be done with replacement, i.e. mostly with
+   REMOVE: Sampling will be done without replacement, i.e. mostly with
+       fd_wsample_sample_and_remove and/or
+       fd_wsample_sample_and_remove_many.
+   NOREMOVE: Sampling will be done with replacement, i.e. mostly with
        fd_wsample_sample and/or
        fd_wsample_sample_many. */
 #define FD_WSAMPLE_HINT_FLAT              0
-#define FD_WSAMPLE_HINT_POWERLAW_NODELETE 2
-#define FD_WSAMPLE_HINT_POWERLAW_DELETE   3
-
-/* fd_wsample_new formats a memory region with the appropriate alignment
-   and footprint to be usable as a stake weight tree.  shmem is a
-   pointer to the first byte of the memory region to use.  rng must be a
-   local join of a ChaCha20 RNG struct. The stake weight tree will use
-   rng to generate random numbers.  The tree owning its own rng seems
-   more natural, but this is done to facilitate sharing of rngs between
-   stake weight trees, which is useful for Turbine.  weights points to
-   the first element of an array of length ele_cnt.  If undelete_enabled
-   is set to 0, fd_wsample_undelete_all will not work but the required
-   footprint is smaller. All elements in weights must be strictly
-   positive, and the sum must be less than ULONG_MAX.  ele_cnt must be
-   less than UINT_MAX.  opt_hint gives a hint of the shape of the
-   weights and the style of queries that will be most common; this hint
-   impacts query performance but not correctness.  opt_hint must be one
-   of FD_WSAMPLE_HINT_*.  Retains read/write interest in rng but not in
-   the memory pointed to by weights.
-
-   On successful return, the tree contains an element corresponding to
-   each provided weight.  Returns shmem on success and NULL on failure.
+#define FD_WSAMPLE_HINT_POWERLAW_NOREMOVE 2
+#define FD_WSAMPLE_HINT_POWERLAW_REMOVE   3
+
+/* fd_wsample_new_init, fd_wsample_new_add_weight, and
+   fd_wsample_new_fini format a memory region with the appropriate
+   alignment and footprint to be usable as a weighted sampler.  This
+   multi-function intitialization process prevents needing to construct
+   a flat array of weights, which is often inconvenient.
+
+   The caller must first call fd_wsample_new_init, then new_add_weight
+   ele_cnt times, and finally new_fini.  Only at that point will the
+   region of memory be ready to be joined.
+
+   fd_wsample_new_init begins the formartting a memory region.
```

Comment:
nit: typo "formatting"

Reply:
Another nit this whole comment could be rewrapped, the line breaking is a bit off.